src/Aspire.Dashboard/Components/ResourcesGridColumns/GridColumnManager.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Components.Resize;
+using Aspire.Dashboard.Model;
+
+namespace Aspire.Dashboard.Components;
+
+public class GridColumnManager : IDisposable
+{
+    private readonly GridColumn[] _columns;
+    private readonly DimensionManager _dimensionManager;
+    private ViewportInformation _currentViewport;
+
+    public GridColumnManager(GridColumn[] columns, ViewportInformation viewportInformation, DimensionManager dimensionManager)
+    {
+        _columns = columns;
+        _currentViewport = viewportInformation;
+        _dimensionManager = dimensionManager;
+        _dimensionManager.OnBrowserDimensionsChanged += OnBrowserDimensionsChanged;
+    }
+
+    private void OnBrowserDimensionsChanged(object _, BrowserDimensionsChangedEventArgs args)
+    {
+        _currentViewport = args.ViewportInformation;
+    }
+
+    public bool IsColumnVisible(string columnName)
```

Comment:
Nit: There is ColumnId on the column component and column name here. Should be consistent.